airflow/ui/openapi-gen/requests/types.gen.ts

```diff
@@ -2159,6 +2158,28 @@ export type PostClearTaskInstancesData = {
 
 export type PostClearTaskInstancesResponse = TaskInstanceCollectionResponse;
 
+export type PatchTaskInstanceDryRunData = {
+  dagId: string;
+  dagRunId: string;
+  mapIndex: number;
+  requestBody: PatchTaskInstanceBody;
+  taskId: string;
+  updateMask?: Array<string> | null;
+};
+
+export type PatchTaskInstanceDryRunResponse = TaskInstanceCollectionResponse;
+
+export type PatchTaskInstanceDryRun1Data = {
```

Comment:
Why does this type exist?

Reply:
That's because we have two endpoints on the same 'controller', the one for classic task instance, and the one for 'maped task' instance. For mapped task instance `map_index` is required and in the path parameter.

For non mapped TI, `map_index` is optional in the query parameters and defaulting to -1.